.github/workflows/build-container.yml

```diff
@@ -11,30 +11,61 @@ jobs:
-
-      - name: Prepare environment
+      - 
+        uses: actions/checkout@v2
+      - 
+        name: Prepare environment
         run: |
           sudo apt-get -qq update
           sudo apt-get -y -o Dpkg::Options::="--force-confnew" install moby-engine moby-cli
           sudo apt-get install -y openssl parallel
-
-      - name: Build container
-        run: |
-          docker swarm init --advertise-addr `ip addr s | grep global | grep -oE '((1?[0-9][0-9]?|2[0-4][0-9]|25[0-5])\.){3}(1?[0-9][0-9]?|2[0-4][0-9]|25[0-5])' | head -n1`
-          docker build --build-arg VERSION="${INSP_VERSION}" -t inspircd:testing .
-
-      - name: Push container
+      -
+        name: Docker meta
         if: github.repository == 'inspircd/inspircd-docker' && github.ref == 'refs/heads/master'
-        run: |
-          if [ "$DOCKER_USERNAME" != "" ] && [ "$DOCKER_PASSWORD" != "" ]; then
-            ./tools/tagging.sh "${DOCKER_REPO_PREFIX}/${DOCKER_REPO_NAME}" "${INSP_VERSION:-latest}" ""
-            docker images "${DOCKER_REPO_PREFIX}/${DOCKER_REPO_NAME}"
-            echo "$DOCKER_PASSWORD" | docker login --username "$DOCKER_USERNAME" --password-stdin
-            docker images --format "{{.Repository}}:{{.Tag}}" "${DOCKER_REPO_PREFIX}/${DOCKER_REPO_NAME}" | xargs -L 1 docker push
-            docker run --rm -v "$(pwd)/:/data/:ro" -e "DOCKERHUB_USERNAME=$DOCKER_USERNAME" -e "DOCKERHUB_PASSWORD=$DOCKER_PASSWORD" -e "DOCKERHUB_REPO_PREFIX=$DOCKER_REPO_PREFIX" -e "DOCKERHUB_REPO_NAME=$DOCKER_REPO_NAME" sheogorath/readme-to-dockerhub
-          fi
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: ${{ env.DOCKER_REPO_PREFIX }}/${{ env.DOCKER_REPO_NAME }}
+          tags: |
+            type=semver,pattern={{version}},value=${{ env.INSP_VERSION }}
+            type=semver,pattern={{major}}.{{minor}},value=${{ env.INSP_VERSION }}
+            type=semver,pattern={{major}},value=${{ env.INSP_VERSION }}
+      - 
+        name: Set up QEMU
+        if: github.repository == 'inspircd/inspircd-docker' && github.ref == 'refs/heads/master'
+        uses: docker/setup-qemu-action@v1
+      -
+        name: Set up Docker Buildx
+        if: github.repository == 'inspircd/inspircd-docker' && github.ref == 'refs/heads/master'
+        uses: docker/setup-buildx-action@v1
+      -
+        name: Login to DockerHub
+        if: github.repository == 'inspircd/inspircd-docker' && github.ref == 'refs/heads/master'
+        uses: docker/login-action@v1 
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      -
+        name: Build and push
+        if: github.repository == 'inspircd/inspircd-docker' && github.ref == 'refs/heads/master'
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          build-args: VERSION=${{ env.INSP_VERSION }}
+          platforms: linux/amd64,linux/arm64
+          push: true
```

Comment:
Won't this try to push and fail if not in the main repository?

Reply:
no - i've added the `if` statements to most of the build steps, so it just does pretty much nothing if you are on a branch. see https://github.com/kaitlinsm/inspircd-docker/runs/2963949767?check_suite_focus=true for what the build does if not on main.

Reply:
It needs to be able to just build in a PR branch without pushing so we can tell if the update was good or not before merging it.

Reply:
+1. You can use the current condition for building to decide whether to push.

Reply:
Ok, so I think that's addressed - I can't test it until it's merged though which is a bit... tricky. It's built amd64 only on my branch, and `push` is set to `false`.